lib/snapshot.go

```diff
@@ -666,7 +666,16 @@ func (snap *Snapshot) FinishProcessBlock(blockNode *BlockNode) {
 
 	snap.CurrentEpochSnapshotMetadata.updateMutex.Lock()
 	defer snap.CurrentEpochSnapshotMetadata.updateMutex.Unlock()
-	if uint64(blockNode.Height)%snap.SnapshotBlockHeightPeriod == 0 &&
+
+	// If the block height is divisible by the snapshot block height period, we update the snapshot metadata.
+	// For PoW blocks, until the first PoS fork height, we use the default snapshot block height period of 1000.
+	// For blocks after the first PoS fork height, we use the snapshot block height period defined in the params.
+	snapshotBlockHeightPeriod := snap.SnapshotBlockHeightPeriod
+	if blockNode.Height < snap.params.ForkHeights.ProofOfStake1StateSetupBlockHeight {
+		snapshotBlockHeightPeriod = snap.params.DefaultPoWSnapshotBlockHeightPeriod
+	}
```

Comment:
small request - can we make a function on params called `GetSnapshotBlockHeightPeriod` that does this conditional logic for fetching the PoS vs. PoW SnapshotBlockHeightPeriod? Will be nice in the event we need to use this in more places.

Reply:
+1 to this. That way, you can reuse it in server.go line 1034